osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs

```diff
@@ -70,8 +91,24 @@ protected override WebRequest CreateWebRequest()
 
             if (cursor != null)
             {
+                if (cursor.Title != null)
+                    req.AddParameter("cursor[title.raw]", cursor.Title);
+                if (cursor.Artist != null)
+                    req.AddParameter("cursor[artist.raw]", cursor.Artist);
+                if (cursor.Difficulty != null)
+                    req.AddParameter("cursor[beatmaps.difficultyrating]", cursor.Difficulty);
+                if (cursor.Ranked != null)
+                    req.AddParameter("cursor[approved_date]", cursor.Ranked);
+                if (cursor.Rating != null)
+                    req.AddParameter("cursor[rating]", cursor.Rating);
+                if (cursor.Plays != null)
+                    req.AddParameter("cursor[play_count]", cursor.Plays);
+                if (cursor.Favourites != null)
+                    req.AddParameter("cursor[favourite_count]", cursor.Favourites);
+                if (cursor.Relevance != null)
+                    req.AddParameter("cursor[_score]", cursor.Relevance);
```

Comment:
I don't think this is a good way to resolve the issue. Consider the possibility of new fields being added to API at some point in time - without game-side changes sorting will be broken for those new fields.

I'm still of the opinion that the game should either not interpret the cursor at all and just send back what it received (which requires API-side changes), or if we decide to handle this game-side, it should be in a way that requires no manual intervention in the future in case new fields are introduced, as long as API still accepts the cursor back the way it currently does (i.e. the field naming convention is still the same).

Reply:
I'm not really sure what I can do about it on my end

For now I can only think of one more way to do it:
* Only `"_id"` parameter would be parsed for all responses instead and second parameter would be taken no matter the property name
* But that presumes that there will be always only 2 cursor fields, which isn't guarantee aswell, so it doesn't sound any more robust

As I said, I'm not really sure as to what I'm expected to do about this one
I may have missed the point again, though

Reply:
The way I'd go forward with this is to throw away the notion of this (or *any*) particular cursor and add some code path that would leverage JSON deserialisation to yield us the JSON object -> query string mapping. I'm not saying it should handle arbitrary JSON object graphs because that's overkill - I guess we can assume the object is a flat object (nothing nested) for now.

So in general I'd like to see something that can, for all requests, turn the cursor part of the response as follows:

```js
{
    // rest of response
    "cursor":
    {
        "foo": "xyz",
        "bar": "baz"
    }
}
```

into this query string:

```
?cursor[foo]=xyz&cursor[bar]=baz
```

*without* interpreting what `foo`, `bar`, `baz` and `xyz` are or mean.

The reason I'm pushing for this is that there will be more cursors for more API requests and if we can save code by not having specialised structures for each requests I'm all for it. In fact some other currently implemented cursors are probably broken but not used (haven't checked myself but I wouldn't be surprised).

Reply:
Would this be fine by you (if I understand correctly)